src/include/duckdb/parser/query_node.hpp

```diff
@@ -23,6 +23,19 @@ enum QueryNodeType : uint8_t {
 	RECURSIVE_CTE_NODE = 4
 };
 
+struct CommonTableExpressionInfo;
+
+class CommonTableExpressionMap {
+public:
+	CommonTableExpressionMap();
+	CommonTableExpressionMap(const CommonTableExpressionMap &other);
```

Comment:
We generally prefer to leave the copy constructor of complex objects deleted to avoid accidentally creating copies, and instead opt for an explicit `Copy()` method.